java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -702,6 +704,241 @@ public static CompletableFuture<Long> arrlen(
+    }
+
+    /**
+     * Retrieves the number of key-value pairs in the object values at the specified <code>path</code>
+     * within the JSON document stored at <code>key</code>.<br>
+     * Equivalent to {@link #objlen(BaseClient, GlideString, GlideString)} with <code>path</code> set
+     * to <code>gs(".")</code>.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @return The object length stored at the root of the document. If document root is not an
+     *     object, an error is raised.<br>
+     *     If <code>key</code> doesn't exist, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\": 1.0, \"b\": {\"a\": {\"x\": 1, \"y\": 2}, \"b\": 2.5, \"c\": true}}").get();
+     * var res = Json.objlen(client, gs("doc"), gs(".")).get();
+     * assert res == 2; // the size of object matching the path `.`, which has 2 keys: 'a' and 'b'.
+     * }</pre>
+     */
+    public static CompletableFuture<Long> objlen(
+            @NonNull BaseClient client, @NonNull GlideString key) {
+        return executeCommand(client, new GlideString[] {gs(JSON_OBJLEN), key});
+    }
+
+    /**
+     * Retrieves the number of key-value pairs in the object values at the specified <code>path</code>
+     * within the JSON document stored at <code>key</code>.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @param path The path within the JSON document.
+     * @return
+     *     <ul>
+     *       <li>For JSONPath (<code>path</code> starts with <code>$</code>):<br>
+     *           Returns an <code>Object[]</code> with a list of long integers for every possible
+     *           path, indicating the number of key-value pairs for each matching object, or <code>
+     *           null
+     *           </code> for JSON values matching the path that are not an object. If <code>path
+     *           </code> does not exist, an empty array will be returned.
+     *       <li>For legacy path (<code>path</code> doesn't start with <code>$</code>):<br>
+     *           Returns the number of key-value pairs for the object value matching the path. If
+     *           multiple paths are matched, returns the length of the first matching object. If
+     *           <code>path</code> doesn't exist or the value at <code>path</code> is not an array, an
+     *           error is raised.
+     *     </ul>
+     *     If <code>key</code> doesn't exist, returns <code>null</code>.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\": 1.0, \"b\": {\"a\": {\"x\": 1, \"y\": 2}, \"b\": 2.5, \"c\": true}}").get();
+     * var res = Json.objlen(client, "doc", ".").get();
+     * assert res == 2; // the size of object matching the path `.`, which has 2 keys: 'a' and 'b'.
+     * res = Json.objlen(client, "doc", "$.b").get();
```

Comment:
whitespace prior.  
Can we add a quick description that the first example is with a legacy path and it returns the first value as a long. The second example uses a JSONPath and returns an array of Long objects.